samza-test/src/test/java/org/apache/samza/test/operator/StreamApplicationIntegrationTestHarness.java

```diff
@@ -226,10 +233,12 @@ public void runApplication(StreamApplication streamApplication, String appName,
     configs.put("systems.kafka.samza.key.serde", "string");
     configs.put("systems.kafka.samza.msg.serde", "string");
     configs.put("systems.kafka.samza.offset.default", "oldest");
+    configs.put("systems.kafka.samza.delete.messages.enabled", "true");
     configs.put("job.coordinator.system", "kafka");
     configs.put("job.default.system", "kafka");
     configs.put("job.coordinator.replication.factor", "1");
     configs.put("task.window.ms", "1000");
+    configs.put("task.checkpoint.factory", TestStreamManager.MockCheckpointManagerFactory.class.getName());
```

Comment:
why is this needed?

Reply:
This is because if we do not define checkpoint factory and if we do not have checkpoint listener, `OffsetManager.buildCheckpoint()` will return null. And if we do not have checkpoint, we will not delete messages in `TaskInstance.commit()`.